### PR TITLE
Get logs from bootstrap node

### PR DESCRIPTION
**Pull request by derekhiggins**

### CODE REVIEW

**Comment by derekhiggins** [comment]
/hold
testing

File: ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh [hunk 1]
```diff
@@ -49,4 +49,12 @@ sosreport --ticket-number "\$HOSTNAME" --batch -o container_log,filesys,kvm,libv
 
 # Get libvirt logs
 tar -czC "/var/log/libvirt/qemu" -f "/tmp/artifacts/libvirt-logs-\$HOSTNAME.tar.gz" --transform "s?^\.?libvirt-logs-\$HOSTNAME?" .
+
+
+# Get the Botstrap logs if its still around, due to an error
+. common.sh
+. network.sh
+ssh -o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' core@\$BOOTSTRAP_PROVISIONING_IP \
+    sudo bash -c \\"ls /var/lib/containers/storage/overlay-containers/*/userdata/ctr.log /var/log/journal/*/system.journal \\| xargs -t tar -czf - \\" > /tmp/artifacts/bootstrap-\$HOSTNAME.tar.gz  || true
```
Comment (stbenjam) [inline ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh:58]: There's an existing script on the bootstrap to collect more logging, I think that's more appropriate. It's /usr/local/bin/installer-gather.sh on the bootstrap.

Source: https://github.com/openshift/installer/blob/45fd42fd0827e7a259043d54027efb3df868cde4/data/data/bootstrap/files/usr/local/bin/installer-gather.sh

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh [hunk 2]
```diff
@@ -89,6 +89,8 @@ cd dev-scripts
 
 cp /root/pull-secret /root/dev-scripts/pull_secret.json
 
+# provoking a failure in the bootstrap
+echo "export ROOT_DISK=/dev/sdz" >> /root/dev-scripts/config_root.sh
```
Comment (stbenjam) [inline ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:93]: This didn't seem to work, e2e-metal-ipi passed

**Review by stbenjam — CHANGES_REQUESTED** [review]
Note the installer has a mechanism that collects the log bundle from the bootstrap. The only time it doesn't work is on terraform failure, which can be fixed in the installer: https://github.com/openshift/installer/issues/3927.  The fix for that is stalled at the moment, so I think this is fine for now.

Ideally we don't collect the logs twice, so we should check if the installer created a log-bundle tarball before running it manually.

File: ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh (hunk 1 above)
Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/devscripts/gather/baremetalds-devscripts-gather-commands.sh:58]: thanks, will update

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh (hunk 2 above)
Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh:93]: hmm, looks like we need to update config_example.sh (will do this), it should be ROOT_DISK_NAME=...

**Comment by derekhiggins** [comment]
/unhold
patch worked, logs retrieved as expected
it doesn't work for the "virtualmedia" job as there is no provisioning ip but its good enough for the moment

**Comment by stbenjam** [comment]
/approve

Thanks! This looks pretty good.  For virtualmedia we could use the API VIP, but there's a good chance in bootstrap failure we never even got far enough to bring it up.

File: ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh (hunk 2 above)
Comment (derekhiggins) [reply ci-operator/step-registry/baremetalds/devscripts/setup/baremetalds-devscripts-setup-commands.sh]: https://github.com/openshift-metal3/dev-scripts/pull/1181

**Pull request merged**